src/tracing/json/json_tracing.cpp

```diff
@@ -295,11 +296,11 @@ void JsonBackend::TraceCounter(const char * label)
     OutputValue(value);
 }
 
-void JsonBackend::TraceMetric(const char * label, int32_t val)
+void JsonBackend::LogEvent(MetricEvent & event)
 {
     ::Json::Value value;
-    value["label"] = label;
-    value["value"] = val;
+    value["label"] = event.key;
+    value["value"] = event.value.store.int32_value;
```

Comment:
is this always guaranteed to be a int32_value ?

Should we then maybe name the event as Int32Metric or similar?

Reply:
From the definition I see value is a union. This code does not handle all possible values and seems to potentially access invalid data.

Reply:
I will handle unsigned type here as well

Reply:
Handled all types using code from https://github.com/andy31415/connectedhomeip/commit/846b3c9984f0fdaed8440ab7d5d9d8451621a919